staging/src/k8s.io/apiserver/pkg/server/egressselector/egress_selector.go

```diff
@@ -203,7 +203,8 @@ func (u *udsGRPCConnector) connect() (proxier, error) {
 		return c, err
 	})
 
-	tunnel, err := client.CreateSingleUseGrpcTunnel(udsName, dialOption, grpc.WithInsecure())
+        ctx := context.TODO()
```

Comment:
couldn't the delta have been limited to this, and defer the proxier change until used?

Reply:
+1

Reply:
This is not possible due to KNP v0.0.21. Due to the breaking change merged in v0.0.21 here: https://github.com/jkh52/apiserver-network-proxy/blob/89eb6ca8c58f2085f1de5e7e699cc0a75d9cd4dc/konnectivity-client/pkg/client/client.go#L178

Reply:
JD points out that ```g.tunnel.Dial("tcp", addr)``` changes to ```return g.tunnel.DialContext(ctx, "tcp", addr)``` so the proxy signature has to change.

Reply:
agreed, thanks.